src/index.ts

```diff
@@ -0,0 +1,353 @@
+          githubClient = new GithubClient(source.platform.url);
+        } else {
+          githubClient = new GithubClient();
+        }
+
+        let profileData = (await githubClient.gql.sendQuery(
+          github.queries.profile(),
+          {
+            username: source.user,
+          },
+          {
+            authorization: source.authorization,
+          }
+        )) as IDataUser;
+
+        let calendarData = (await githubClient.gql.sendQuery(
+          github.queries.calendar(profileData.data.user as IProfile),
+          {
+            username: source.user,
+          },
+          {
+            authorization: source.authorization,
+          }
+        )) as IDataUser;
+
+        const data: IGitHubData = {
+          profile: profileData.data.user,
+          calendar: calendarData.data.user,
+        };
+
+        await github.converter.run(data);
+      } else if (platform === "gitlab") {
+        let gitlabClient: WebClient;
+
+        // Init gitlab client
+        /* Use the default client url if none is provided */
+        if (source.platform.url) {
+          gitlabClient = new WebClient(source.platform.url);
+        } else {
+          gitlabClient = new WebClient("https://gitlab.com");
+        }
+
+        let rawData = {
+          platform: {
+            name: source.platform.name,
+            url: source.platform.url
+              ? source.platform.url
+              : "https://gitlab.com",
+          },
+          home: await gitlabClient.scraper.getDom(
+            gitlab.paths.home(source.user)
+          ),
+          atom: await gitlabClient.scraper.getDom(
+            gitlab.paths.atom(source.user)
+          ),
+          currentCalendar: await gitlabClient.scraper.getJson<{
+            [index: string]: number;
+          }>(gitlab.paths.currentCalendar(source.user)),
+          groups: await gitlabClient.scraper.getJson<{ html: string }>(
+            gitlab.paths.groups(source.user)
+          ),
+          projects: await gitlabClient.scraper.getJson<{
+            html: string;
+          }>(gitlab.paths.projects(source.user)),
+          activity: await gitlabClient.scraper.getJson<{
+            html: string;
+          }>(gitlab.paths.activity(source.user)),
+        };
+
+        gitlab.converter.runScraper(rawData);
+      }
+    } catch (err) {
+      //#LEGACY
+      console.error("INTEL GITLAB", err);
+    }
+  }
+
+  /**
+   * Append a list of source objects.
+   *
+   * @param sources A list of source objects
+   * @description Calls .append() for each source object in the provided list
+   */
+  async appendList(sources: ISource[]) {
+    /* Sources which platform name is github should be before others */
+    sources = sources.sort((a, b) => {
+      const platformNameOrder = ["github", "gitlab"];
+
+      const aPlatformNameIndex = platformNameOrder.indexOf(a.platform.name);
+      const bPlatformNameIndex = platformNameOrder.indexOf(b.platform.name);
```

Comment:
For all reviewers:
This newline is fully acceptable due to logical separation.

Reply:
true